model_runner/validator/validator_utils.py

```diff
@@ -0,0 +1,126 @@
+# input: path to settings file (e.g. .json file)
+#   1. path to data --> test: string, path exists OK
+#   2. program to run --> test: string, file exists OK
+#   3. settings for job array
+#     * #gpus --> test: int OK
+#     * gpu type --> test: str, in list of existing GPUs OK
+#     * #processor cores: --> test: int OK
+#     * memory per processor core --> test: int OK
+#     * scratch per processor core --> test: int OK
+#     * run_time --> test: str or int, either in min or [h] format OK
+#     * #jobs to run --> test: int OK
+#     * #jobs to run in parallel --> test: int OK
+#     * log-file name template --> test: str OK
+#   4. hyper-parameter margins (+dtype) --> test: Dict[str, List[Any]], OK
+# output: None
+
+# NOTE: for now the hyper-parameter grid has to be submitted explicitly
+# TODO: later patch explicit version to also accomodate continuous ranges, discrete variables, ...
+# TODO: include feature to check if required resources can be allocated to a single machine or require multiple machines
+# TODO: set up tests for input function in _tests
+# TODO: include type coercion
+
+import os
+from typing import Any, Dict, List, Union
+
+from pydantic import BaseModel, validator
+
+
+class JobArrayModel(BaseModel):
+    run_time: Union[str, int]
+    processor_cores: int
+    memory: int
+    scratch: int
+    gpu_type: str
+    ngpus: int
+    njobs: int
+    njobs_parallel: int
+    logfile_name_template: str
+
+    @validator("gpu_type")
+    def validate_gpu_type(cls, v):
+        """
+        Validates if submitted gpu_type is currently available at Euler
+        (https://scicomp.ethz.ch/wiki/Getting_started_with_clusters#GPU).
+        """
+        gpu_list = [
+            "GeForceGTX1080",
+            "GeForceGTX1080Ti",
+            "GeForceRTX2080Ti",
+            "GeForceRTX2080Ti",
+            "TITANRTX",
+            "TeslaV100_SXM2_32GB",
+            "TeslaV100_SXM2_32GB",
+            "A100_PCIE_40GB",
+        ]
+
+        if v not in gpu_list:
+            raise ValueError(
+                f'"{v}" is currently not available on Euler. Check the list of available GPUs '
+                "at https://scicomp.ethz.ch/wiki/Getting_started_with_clusters#GPU."
+            )
+
+        return v
+
+    @validator("run_time")
+    def validate_and_coerce_run_time(cls, v):
+        """
+        Validates if submitted run_time adheres to the format specified in
+        https://scicomp.ethz.ch/wiki/Getting_started_with_clusters#Wall-clock_time.
+        """
+        if type(v) == str:
```

Comment:
I think this is fine, but in the future, we could consider using the datetime module for validating the time.

Reply:
From what I [see](https://docs.python.org/3/library/time.html#time.strftime) `datetime` or `time` modules may not be applicable/beneficial (i.e. `time.strptime('120:00', '%H:%M')` would throw an `ValueError: time data '120:00' does not match format '%H:%M'`, as >24h are not accepted).

Reply:
Thanks for looking into it, @dsethz . I agree that we  should move forward validator code you've written.